src/commands/plugins.rs

```diff
@@ -242,11 +243,99 @@ impl Upgrade {
 
         if self.all {
             self.upgrade_all(manifests_dir).await
+        } else if self.name.is_none()
+            && self.local_manifest_src.is_none()
+            && self.remote_manifest_src.is_none()
+        {
+            // Default behavior (multiselect)
+            self.upgrade_multiselect().await
         } else {
             self.upgrade_one().await
         }
     }
 
+    // Multiselect plugin upgrade experience
+    async fn upgrade_multiselect(self) -> Result<()> {
+        let catalogue_plugins = list_catalogue_plugins().await?;
+        let installed_plugins = list_installed_plugins()?;
+
+        let installed_in_catalogue: Vec<_> = installed_plugins
+            .into_iter()
+            .filter(|installed| {
+                catalogue_plugins.iter().any(|catalogue| {
+                    installed.manifest == catalogue.manifest
+                        && matches!(catalogue.compatibility, PluginCompatibility::Compatible)
```

Comment:
I am not sure we need this check.  We don't really care if the catalogue copy of the installed plugin is compatible.  In fact, a possible case is:

* User installs Plugin 1.0 from catalogue.  1.0 is compatible with Spin 3.0 or below.
* User upgrades to Spin 4.1.
* User tries to run Plugin and it warns about compatibility.
* User tries to upgrade.

In this case we would _not_ want to filter Plugin 1.0 out of the list.

We care about compatibility on _upgrade versions_ but I don't think that's what this is looking at.  I might be misunderstanding though?  (As you've rightly noted, unit tests would be the way to catch cases like this - it's hard to simulate with actual plugins!)

Reply:
To check the compatibility of the _upgrade versions_, I believe using the `PluginCompatibility::for_current()` is the best way to do it. Correct me, if I'm wrong.

So, I'll add the condition below after the versions comparison:

`if let PluginCompatibility::Compatible = PluginCompatibility::for_current(&manifest)`

Reply:
That sounds good - thanks!

As a style detail, `if let PluginCompatibility::Compatible = PC::for_current(...)` might be more idiomatic as `matches!(PC::for_current(...), PC::Compatible)` - the `let` is a bit odd because no actual assignment is possible.  You could also add `PartialEq` to the PluginCompatibility `#[derive(...)]` list, which would allow you to write `if PC::Compatible == PC::for_current`.  None of this is a blocker though!

Reply:
You're totally right. Thanks for the hint!